SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -439,38 +439,76 @@ void HcalTriggerPrimitiveAlgo::analyzeQIE11(IntegerCaloSamples& samples,
   for (unsigned int ibin = 0; ibin < dgSamples - shrink; ++ibin) {
     int algosumvalue = 0;
     bool check_sat = false;
-    for (unsigned int i = 0; i < filterSamples; i++) {
+    //TP energy calculation for PFA2
+    if (weightsQIE11_[theIeta][0] == 255) {
+      for (unsigned int i = 0; i < filterSamples; i++) {
+        //add up value * scale factor
+        // In addition, divide by two in the 10 degree phi segmentation region
+        // to mimic 5 degree segmentation for the trigger
+        unsigned int sample = samples[ibin + i];
+
+        if (fix_saturation_ && (sample_saturation.size() > ibin + i))
+          check_sat = (check_sat | sample_saturation[ibin + i] | (sample > QIE11_MAX_LINEARIZATION_ET));
+
+        if (sample > QIE11_MAX_LINEARIZATION_ET)
+          sample = QIE11_MAX_LINEARIZATION_ET;
+
+        // Usually use a segmentation factor of 1.0 but for ieta >= 21 use 0.5
```

Comment:
This comment is outdated with the new definition of `segmentationFactor`

Reply:
Thanks @perrotta for all of the comments! I have implemented them